### PR TITLE
refac: added support for B2B error acceptance tests and refactored TestRunner into a collection

### PR DESCRIPTION
**Pull request by andpou**

### CODE REVIEW

File: source/AcceptanceTests/Factories/PayloadBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿using System.IO.Pipes;
+using System.Runtime.InteropServices;
+using System.Xml;
+
+namespace EndToEndTesting.Requests;
+
+public class PayloadBuilder
+{
+
+    public PayloadBuilder()
```
[the remark this answers is not in the document]
Comment (andpou) [reply source/AcceptanceTests/Factories/PayloadBuilder.cs]: Aadded

File: source/IntegrationTests/Application/Transactions/AggregatedMeasureData/WhenAnAcceptedResultIsAvailableTests.cs [hunk 2]
```diff
@@ -29,6 +29,7 @@
 using Energinet.DataHub.EDI.Process.Domain.Transactions.Aggregations.OutgoingMessage;
 using Energinet.DataHub.EDI.Process.Infrastructure.Configuration.DataAccess;
 using Energinet.DataHub.Edi.Responses;
+using Energinet.DataHub.MeteringPoints.RequestResponse.Response;
```
[the remark this answers is not in the document]
Comment (andpou) [reply source/IntegrationTests/Application/Transactions/AggregatedMeasureData/WhenAnAcceptedResultIsAvailableTests.cs:32]: I should not have changes in this file

File: source/AcceptanceTests/Tests/B2BErrors/WhenPayloadDataIsDifferentFromTokenData.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+using Energinet.DataHub.EDI.AcceptanceTests.TestData;
+using Energinet.DataHub.EDI.AcceptanceTests.Tests.Asserters;
+using Xunit.Abstractions;
+
+namespace Energinet.DataHub.EDI.AcceptanceTests.Tests.B2BErrors;
+
+[SuppressMessage("Reliability", "CA2007:Consider calling ConfigureAwait on the awaited task", Justification = "Test code should not configure await.")]
+[SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "Testing")]
+[Collection("Acceptance test collection")]
+public class WhenPayloadDataIsDifferentFromTokenData
```
[the remark this answers is not in the document]
Comment (andpou) [reply source/AcceptanceTests/Tests/B2BErrors/WhenPayloadDataIsDifferentFromTokenData.cs]: gemmer til senere som team snak

File: source/AcceptanceTests/Factories/PayloadBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
+﻿using System.IO.Pipes;
```
[the remark this answers is not in the document]
Comment (andpou) [reply source/AcceptanceTests/Factories/PayloadBuilder.cs]: added

File: source/AcceptanceTests/Tests/B2BErrors/WhenPayloadDataIsDifferentFromTokenData.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    private readonly AggregatedMeasureDataRequestDsl _aggregationRequest;
+
+    public WhenPayloadDataIsDifferentFromTokenData(ITestOutputHelper output, TestRunner runner)
+    {
+        _output = output;
+        _aggregationRequest = _aggregationRequest = new AggregatedMeasureDataRequestDsl(new EdiDriver(runner.AzpToken, runner.ConnectionString));
+    }
+
+    [Fact]
+    public async Task SenderMarketParticipantMridIsDifferentFromMridInTokenAsync()
```
[the remark this answers is not in the document]
Comment (andpou) [reply source/AcceptanceTests/Tests/B2BErrors/WhenPayloadDataIsDifferentFromTokenData.cs]: fixed

**Review by MadsDue — APPROVED** [review]

**Pull request merged**